transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -759,8 +760,10 @@ protected void flush0() {
             try {
                 doWrite(outboundBuffer);
             } catch (Throwable t) {
-                outboundBuffer.failFlushed(t);
-                if (t instanceof IOException && config().isAutoClose()) {
+                boolean close = t instanceof IOException && config().isAutoClose();
+                // We not want to notify for pendingWritableBytes updates if we close the channel anyway.
```

Comment:
'We do not want to trigger channelWritabilityChanged event if the channel is going to be closed.`